packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx

```diff
@@ -66,6 +67,14 @@ export const AnnounceCandidacyModal = () => {
     }
   }, [state, activeMember?.id, JSON.stringify(feeInfo), hasRequiredStake])
 
+  useEffect((): any => {
+    if (state.matches('staking') && state.context.stakingAccount && state.context.stakingAmount) {
+      return setValidNext(true)
+    }
+
+    return setValidNext(false)
+  }, [state, activeMember?.id, stakingStatus])
```

Comment:
Using return statements there confused me for a hot minute, because my brain immediately started looking for a cleanup function. This is syntactically correct but I'd use an `else` clause myself instead of escaping via `return`.

Reply:
Yeah, this might be confusing. So from me :+1: